core/orchestrator.go

```diff
@@ -320,17 +322,25 @@ func (n *LivepeerNode) transcodeSeg(config transcodeConfig, seg *stream.HLSSegme
 	start := time.Now()
 	tData, err := transcoder.Transcode(url, md.Profiles)
 	if err != nil {
-		glog.Errorf("Error transcoding seg: %v - %v", seg.Name, err)
+		glog.Errorf("Error transcoding manifest=%s segNo=%d segName=%s - %v", string(md.ManifestID), seg.SeqNo, seg.Name, err)
 		return terr(err)
 	}
 	// transcodeEnd := time.Now().UTC()
 	if len(tData) != len(md.Profiles) {
-		glog.Errorf("Did not receive the correct number of transcoded segments; got %v expected %v", len(tData), len(md.Profiles))
+		glog.Errorf("Did not receive the correct number of transcoded segments; got %v expected %v manifest=%s seqNo=%d", len(tData),
+			len(md.Profiles), string(md.ManifestID), seg.SeqNo)
 		return terr(fmt.Errorf("MismatchedSegments"))
 	}
+	for i, segData := range tData {
+		if len(segData) == 0 {
```

Comment:
When does this happen? If we do keep this type of check, maybe better to do it when we also iterate over `md.Profiles`

Reply:
Originally I've hit this in #845. I've spent some time figuring this out, so I've thought we should keep this check.
Why it's better to check `md.Profiles`? It's just checking that there is some data, not that it's aligned with `profiles`.

Reply:
Yeah, it doesn't really hurt to have the check, but I guess that brings up the question why is a length of 0 special as opposed to some other reasonably small minimum number.

> Why it's better to check md.Profiles? It's just checking that there is some data, not that it's aligned with profiles.

So we don't iterate over the results multiple times. We already check whether `tData[i] == nil` so might as well add a ` || len(tData[i]) < minimum` here. And probably returning an error instead of continuing.

Reply:
OK, what minimum size it could be? What the minimum size of valid but empty segment?

Reply:
Made fix

Reply:
In unit tests transcoder returns segment with length 25 - is it ok?

Reply:
Sure, I can't think of a better number either.